src/main/scala/com/microsoft/hyperspace/index/IndexLogManager.scala

```diff
@@ -99,6 +99,10 @@ class IndexLogManagerImpl(indexPath: Path, hadoopConfiguration: Configuration =
           val entry = getLog(id)
           if (entry.isDefined && Constants.STABLE_STATES.contains(entry.get.state)) {
             return entry
+          } else if (entry.isDefined && (entry.get.state.equals(Constants.States.CREATING) ||
+                     entry.get.state.equals(Constants.States.VACUUMING))) {
+            // Not consider unrelated logs before creating or vacuuming state.
+            return None
           }
```

Comment:
Can the following be simplified now?

https://github.com/microsoft/hyperspace/blob/c900cf56997c42873358e1260050a322f8fe5a0c/src/main/scala/com/microsoft/hyperspace/actions/CancelAction.scala#L44-L53